### PR TITLE
Add a tool to convert port alias when used as bash pipe

### PR DESCRIPTION
**Pull request by taoyl-ms**
Sample Usage (on a S6000):

```
admin@s6000-1:~$ portstat                
      Iface    STATE    RX_OK    RX_BPS    RX_UTIL    RX_ERR    RX_DRP    RX_OVR    TX_OK    TX_BPS    TX_UTIL    TX_ERR    TX_DRP    TX_OVR
-----------  -------  -------  --------  ---------  --------  --------  --------  -------  --------  ---------  --------  --------  --------
  Ethernet0        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
  Ethernet4        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
  Ethernet8        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 Ethernet12        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 Ethernet16        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 Ethernet20        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 Ethernet24        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 Ethernet28        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 Ethernet32        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 Ethernet36        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 Ethernet40        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 Ethernet44        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 Ethernet48        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 Ethernet52        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 Ethernet56        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 Ethernet60        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 Ethernet64        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 Ethernet68        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 Ethernet72        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 Ethernet76        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 Ethernet80        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 Ethernet84        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 Ethernet88        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 Ethernet92        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 Ethernet96        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
Ethernet100        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
Ethernet104        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
Ethernet108        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
Ethernet112        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
Ethernet116        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
Ethernet120        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
Ethernet124        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
admin@s6000-1:~$ portstat | port2alias 
      Iface    STATE    RX_OK    RX_BPS    RX_UTIL    RX_ERR    RX_DRP    RX_OVR    TX_OK    TX_BPS    TX_UTIL    TX_ERR    TX_DRP    TX_OVR
-----------  -------  -------  --------  ---------  --------  --------  --------  -------  --------  ---------  --------  --------  --------
  fortyGigE0/0        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
  fortyGigE0/4        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
  fortyGigE0/8        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 fortyGigE0/12        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 fortyGigE0/16        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 fortyGigE0/20        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 fortyGigE0/24        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 fortyGigE0/28        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 fortyGigE0/32        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 fortyGigE0/36        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 fortyGigE0/40        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 fortyGigE0/44        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 fortyGigE0/48        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 fortyGigE0/52        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 fortyGigE0/56        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 fortyGigE0/60        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 fortyGigE0/64        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 fortyGigE0/68        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 fortyGigE0/72        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 fortyGigE0/76        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 fortyGigE0/80        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 fortyGigE0/84        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 fortyGigE0/88        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 fortyGigE0/92        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
 fortyGigE0/96        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
fortyGigE0/100        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
fortyGigE0/104        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
fortyGigE0/108        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
fortyGigE0/112        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
fortyGigE0/116        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
fortyGigE0/120        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
fortyGigE0/124        U        0       N/A        N/A         0         0         0    20748       N/A        N/A         0         0         0
admin@s6000-1:~$
```

Depends on https://github.com/Azure/sonic-buildimage/pull/1072

### CODE REVIEW

File: scripts/port2alias [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    return (platform, hwsku)
+
+def main():
+    (platform, hwsku) = get_platform_hwsku()
+    (port, _) = get_port_config(hwsku, platform)
+    for line in sys.stdin:
+        start = 0
+        end = 0
+        while end < len(line):
+            if line[end].isalnum():
```
Comment (lguohan) [inline scripts/port2alias]: here, you have an assumption that the port name does not contain char other than alnum. For example, the port name cannot be et5_1

File: scripts/port2alias [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        tokens = line.split()
+        if not tokens:
+            continue
+        if tokens[0].lower() == 'hwsku:':
+            hwsku = tokens[1]
+        elif tokens[0].lower() == 'platform:':
+            platform = tokens[1]
+    return (platform, hwsku)
+
+def main():
```
Comment (lguohan) [inline scripts/port2alias]: it is probably good to have a unit test for this string conversion. We need to have some positive and negative case.

for example,

Ethernet3 should not be converted if Ethernet3 is not in the alias map.

Ethernet43 should not be converted if Etherent4 is in but not Ethernet43,

Ethernet4: should be converted if Ethernet4 is in.

File: scripts/port2alias [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    for line in sys.stdin:
+        start = 0
+        end = 0
+        while end < len(line):
+            if line[end].isalnum():
+                pass
+            else: 
+                # End of a word
+                word = line[start:end]
+                if port.has_key(word):
```
Comment (jleveque) [inline scripts/port2alias]: Consider `if word in port:` (or `if word in port.keys():` to make it more understandable at first glance) instead. More portable for the future as it appears has_key() is deprecated and has been removed in Python 3.

File: scripts/port2alias (hunk 1 above)
Comment (taoyl-ms) [reply scripts/port2alias]: What's the legal charset for sonic interface names? I thought only alnum characters are allowed.

**Comment by taoyl-ms** [comment]
retest this please

**Comment by taoyl-ms** [comment]
retest this please

**Comment by taoyl-ms** [comment]
retest this please

**Comment by taoyl-ms** [comment]
retest this please

**Comment by taoyl-ms** [comment]
retest this please

**Comment by taoyl-ms** [comment]
retest this please

**Comment by taoyl-ms** [comment]
retest this please

**Pull request merged**